scipy/interpolate/predict.py

```diff
@@ -0,0 +1,114 @@
+""" This module implements the Always Stable Projector from Ref. [1] that can
+be used for predicting values in a (time) series, such as in [2-3].
+
+[1] J. Kolafa, J. Comput. Chem. 25, 335 (2004).
+[2] T. D. Kuehne, M. Krack, F. Mohamed, and M. Parrinello,
+    Phys. Rev. Lett. 98, 066401 (2007).
+[3] T. Spura, H. Elgabarty, and T. D. Kuehne, Phys. Chem. Chem. Phys. (2015).
+"""
+
+import numpy as np
+import scipy.misc as scimisc
+
+__all__ = ["asp", "Cache"]
+
+
+def asp(array):
+    """
+    Run the Always Stable Projector algorithm.
+
+    Parameters
+    ----------
+    array : array_like
```

Comment:
don't name it ``array``, that's confusing when reading the code (many would think it's ``numpy.array``). ``data`` is fine. or even ``x``.